### PR TITLE
 Sbt play scalajs v0.2.x

### PR DESCRIPTION
**Pull request by vmunier**
* Uses build.sbt instead of Build.scala.
* Several Scala.js projects can be attached to a single Play! project.
* sbt-play-scalajs does not depend on Play anymore. It depends on sbt-web instead which is also used by the Play framework.
* Scala files for Source Maps are copied into the assets folder and source map paths are rewritten. There is no need for a special SourceMaps controller to serve the Scala files anymore.
* Source Maps is disabled in production by default (i.e. when running play start, stage, dist).
  To enable Source Maps in production, `emitSourceMaps in fullOptJS := true` should be set in the Scala.js projects.
* Uses the play-scalajs-scripts library which is used in main.scala.html to select the fast optimized or full optimized javascript file depending on whether the app is running in dev or prod.
* In this example the full optimized Scala.js files are also gzipped.

### CODE REVIEW

**Comment by SRGOM** [comment]
Thank you for your project.
This might seem tangential to the pull request, but may I suggest using play name for jvm and something longer for js, say: scalascript? I feel that would be less confusing. Also, is there a way to stop the creation of java dirs inside the js directory?

**Comment by vmunier** [comment]
Hi @SRGOM 
It's always difficult to find good names. I thought of three directories:
- `example-server` Play app
- `example-client` Scala.js
- `example-shared` Shared code

On how to stop the creation of the java folder: the js/src/main/java directory gets created when you use the play run command because jvm aggregates the js project and watches unmanagedSourceDirectories from the js project.
Use `unmanagedSourceDirectories in Compile := Seq((scalaSource in Compile).value)` in the js settings to prevent the creation of the java dir.

**Comment by austinrivas** [comment]
great work :+1:

File: README.md [hunk 1]
```diff
@@ -38,5 +39,4 @@ $ open http://localhost:9000
 
 ### IntelliJ
 
-1. `$ sbt gen-idea`
-2. Inside IntelliJ, `File/Open...`, choose the root folder to import all the projects (do *not* use `Import Project...` or `Import Module...`)
+IntelliJ integration does not work well with cross projects for now. The IDEA team is working on a fix.
```
Comment (dwijnand) [inline README.md]: Have you noticed it working better now that https://youtrack.jetbrains.com/issue/SCL-7859 has been fixed? Or is it something else?

Comment (vmunier) [inline README.md]: Yep it's working now with the latest version of the Scala Plugin (v1.3.3). The README.md has been updated by the way https://github.com/vmunier/play-with-scalajs-example/commit/939c7f6d702b8b3f698fc6253413e0b9773a5d38 ;-)

Comment (dwijnand) [inline README.md]: Nice.

**Pull request merged**